roles/reproducer/tasks/push_code.yml

```diff
@@ -125,7 +125,7 @@
       when:
         - item.src is abs or item.src is not match('.*:.*')
       ansible.builtin.command:  # noqa: command-instead-of-module
-        cmd: "rsync -ar {{ item.src }} zuul@controller-0:{{ item.dest }}"
+        cmd: "rsync {{ cifmw_push_code_extra_ssh_options | default('') }} -ar {{ item.src }} zuul@controller-0:{{ item.dest }}"
```

Comment:
sounds safer than switching to the module + use _ssh_args.
Adding that option in the doc would be nice.
Also, I don't know if it's still the usage, but the parameter should be named differently, something like `cifmw_reproducer_push_code_ssh_opts` (referencing the role name as prefix).

You can also avoid the `|default('')` by adding it to the defaults/main.yml directly (with associated doc in the README).

Reply:
will do after initial testing